api/cpp/cbindgen.rs

```diff
@@ -49,6 +49,57 @@ fn enums(path: &Path) -> anyhow::Result<()> {
     Ok(())
 }
 
+fn builtin_structs(path: &Path) -> anyhow::Result<()> {
+    let mut structs = std::fs::File::create(path.join("slint_builtin_structs_internal.h"))
+        .context("Error creating slint_builtin_structs_internal.h file")?;
+    writeln!(structs, "#pragma once")?;
+    writeln!(structs, "// This file is auto-generated from {}", file!())?;
+    writeln!(structs, "namespace slint::cbindgen_private {{")?;
```

Comment:
Before this line, you also need to include the header for generated enums, i think